### PR TITLE
Personal/yufei/encryption

### PR DESCRIPTION
**Pull request by moria97**
Add AES encryption method support.

### CODE REVIEW

**Comment by sowu880** [comment]
More encryption related tests should be added in "Fhir.Anonymizer.Core.UnitTests.Visitors"

File: src/Fhir.Anonymizer.Core/AnonymizationConfigurations/AnonymizerConfigurationValidator.cs [hunk 1]
```diff
@@ -38,6 +40,38 @@ public void Validate(AnonymizerConfiguration config)
...
+
+        }
+
+        // The following method takes a bit length input and returns whether that length is a valid size
+        // validSizes for AES: MinSize=128, MaxSize=256, SkipSize=64
+        private bool IsValidKeySize(int bitLength, KeySizes[] validSizes)
+        {
+            int i, j;
+
+            if (validSizes == null) return false;
```
Comment (Patrick2011) [inline src/Fhir.Anonymizer.Core/AnonymizationConfigurations/AnonymizerConfigurationValidator.cs:63]: Use parentheses to make clauses in an expression apparent.

File: src/Fhir.Anonymizer.Core/AnonymizationConfigurations/AnonymizerConfigurationValidator.cs [hunk 2]
```diff
@@ -38,6 +40,38 @@ public void Validate(AnonymizerConfiguration config)
                     throw new AnonymizerConfigurationErrorsException($"{method} not support.");
                 }
             }
+
+            // Check AES key size is valid (16, 24 or 32 bytes).
+            if (!string.IsNullOrEmpty(config.ParameterConfiguration?.EncryptKey))
+            {
+                using Aes aes = Aes.Create();
```
Comment (Patrick2011) [inline src/Fhir.Anonymizer.Core/AnonymizationConfigurations/AnonymizerConfigurationValidator.cs:47]: is there a following clause expected?

File: src/Fhir.Anonymizer.Core/Processors/EncryptProcessor.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        private readonly byte[] _key;
+        private readonly ILogger _logger = AnonymizerLogging.CreateLogger<EncryptProcessor>();
+
+        public EncryptProcessor(string encryptKey)
+        {
+            _key = Encoding.UTF8.GetBytes(encryptKey);
+        }
+
+        public ProcessResult Process(ElementNode node)
+        {
```
Comment (BoyaWu10) [inline src/Fhir.Anonymizer.Core/Processors/EncryptProcessor.cs:20]: Just curious can we configure entrypt for every kinds of data types?

File: src/Fhir.Anonymizer.Core/AnonymizationConfigurations/AnonymizerConfigurationValidator.cs (hunk 1 above)
Comment (moria97) [reply src/Fhir.Anonymizer.Core/AnonymizationConfigurations/AnonymizerConfigurationValidator.cs]: Thanks. Updated in the newest change.

File: src/Fhir.Anonymizer.Core/AnonymizationConfigurations/AnonymizerConfigurationValidator.cs (hunk 2 above)
Comment (moria97) [reply src/Fhir.Anonymizer.Core/AnonymizationConfigurations/AnonymizerConfigurationValidator.cs:47]: yes. this is a simplified using statement here and dispose will happen at the end of scope of the variable aes.

File: src/Fhir.Anonymizer.Core/Processors/EncryptProcessor.cs (hunk 3 above)
Comment (moria97) [reply src/Fhir.Anonymizer.Core/Processors/EncryptProcessor.cs:20]: Yes. We encrypt all fields according to configuration file and the author will decide the format compatibility issues by himself.

Comment (moria97) [reply src/Fhir.Anonymizer.Core/Processors/EncryptProcessor.cs:20]: We will also address this in the document that the encrypt method is better to process field without format constraints

Comment (BoyaWu10) [reply src/Fhir.Anonymizer.Core/Processors/EncryptProcessor.cs:20]: Got it. Thanks!

**Pull request merged**